Screenshot crash with locator mask

Hello, I attempt to screenshot the page and mask the sensitive information by set the Mask property with locator as below code. 

```go
l := page.Locator("//*[@id=\"inputPassword\"]")
po := playwright.PageScreenshotOptions{ 
    Path: playwright.String("test1.png"),
}
po.Mask = append(po.Mask, l)
page.Screenshot(po)
```

But I receive the panic error as below.

```bash
panic: reflect.Value.Interface: cannot return value obtained from unexported field or method

goroutine 1 [running]:
reflect.valueInterface({0x11cf3e0?, 0xc000308300?, 0x1255900?}, 0x1?)
        D:/Program Files/Go/src/reflect/value.go:1491 +0xd8
reflect.Value.Interface(...)
        D:/Program Files/Go/src/reflect/value.go:1480
github.com/playwright-community/playwright-go.transformStructIntoMapIfNeeded({0x11cc5a0, 0xc000308300})
        C:/Users/lonka/go/pkg/mod/github.com/playwright-community/playwright-go@v0.3800.0/helpers.go:73 +0x4cf
github.com/playwright-community/playwright-go.transformStructValues({0x11cc5a0, 0xc000308300?})
        C:/Users/lonka/go/pkg/mod/github.com/playwright-community/playwright-go@v0.3800.0/helpers.go:36 +0x226
github.com/playwright-community/playwright-go.transformStructValues({0x114d3c0, 0xc0001b6170?})
        C:/Users/lonka/go/pkg/mod/github.com/playwright-community/playwright-go@v0.3800.0/helpers.go:42 +0x2ce
github.com/playwright-community/playwright-go.transformStructIntoMapIfNeeded({0x11bb860, 0xc0001b6150})
        C:/Users/lonka/go/pkg/mod/github.com/playwright-community/playwright-go@v0.3800.0/helpers.go:73 +0x4d4
github.com/playwright-community/playwright-go.transformOptions({0xc00011dcb8, 0xc0001ac410?, 0xd0?})
        C:/Users/lonka/go/pkg/mod/github.com/playwright-community/playwright-go@v0.3800.0/helpers.go:123 +0x1fc
github.com/playwright-community/playwright-go.(*channel).innerSend(0xc0002b06c0, {0x11d3b14, 0xa}, 0x0, {0xc00011dcb8?, 0xc000192940?, 0xc000192960?})
        C:/Users/lonka/go/pkg/mod/github.com/playwright-community/playwright-go@v0.3800.0/channel.go:28 +0x5f
github.com/playwright-community/playwright-go.(*channel).Send.func1()
        C:/Users/lonka/go/pkg/mod/github.com/playwright-community/playwright-go@v0.3800.0/channel.go:17 +0x33
github.com/playwright-community/playwright-go.(*connection).WrapAPICall(0xc0000c0280, 0xc00011dc20, 0x20?)
        C:/Users/lonka/go/pkg/mod/github.com/playwright-community/playwright-go@v0.3800.0/connection.go:139 +0xb1
github.com/playwright-community/playwright-go.(*channel).Send(0xc0001b60e0?, {0x11d3b14?, 0x1?}, {0xc00011dcb8?, 0x1449120?, 0xc0001b60e0?})
        C:/Users/lonka/go/pkg/mod/github.com/playwright-community/playwright-go@v0.3800.0/channel.go:16 +0x70
github.com/playwright-community/playwright-go.(*pageImpl).Screenshot(0xc000100640, {0xc0001b60e0?, 0x5?, 0x0?})
        C:/Users/lonka/go/pkg/mod/github.com/playwright-community/playwright-go@v0.3800.0/page.go:324 +0x85
github.com/Lonka/test/main.(*PopulateTextField).Run(0xc0000be000)
        D:/Projects/test/main.go:107 +0x448
main.main()
        D:/Projects/test/main.go:32 +0x18f
```

Is there something I haven't set correctly?
thanks

--- a/element_handle.go
+++ b/element_handle.go
@@ -226,10 +226,29 @@ func (e *elementHandleImpl) SelectText(options ...ElementHandleSelectTextOptions
 
 func (e *elementHandleImpl) Screenshot(options ...ElementHandleScreenshotOptions) ([]byte, error) {
 	var path *string
-	if len(options) > 0 {
+	overrides := map[string]interface{}{}
+	if len(options) == 1 {
 		path = options[0].Path
+		options[0].Path = nil
+		if options[0].Mask != nil {
+			masks := make([]map[string]interface{}, 0)
+			for _, m := range options[0].Mask {
+				if m.Err() != nil { // ErrLocatorNotSameFrame
+					return nil, m.Err()
+				}
+				l, ok := m.(*locatorImpl)
+				if ok {
+					masks = append(masks, map[string]interface{}{
+						"selector": l.selector,
+						"frame":    l.frame.channel,
+					})
+				}
+			}
+			overrides["mask"] = masks
+			options[0].Mask = nil
+		}
 	}
-	data, err := e.channel.Send("screenshot", options)
+	data, err := e.channel.Send("screenshot", options, overrides)
 	if err != nil {
 		return nil, fmt.Errorf("could not send message :%w", err)
 	}

--- a/page.go
+++ b/page.go
@@ -318,10 +318,29 @@ func (p *pageImpl) Request() APIRequestContext {
 
 func (p *pageImpl) Screenshot(options ...PageScreenshotOptions) ([]byte, error) {
 	var path *string
-	if len(options) > 0 {
+	overrides := map[string]interface{}{}
+	if len(options) == 1 {
 		path = options[0].Path
+		options[0].Path = nil
+		if options[0].Mask != nil {
+			masks := make([]map[string]interface{}, 0)
+			for _, m := range options[0].Mask {
+				if m.Err() != nil { // ErrLocatorNotSameFrame
+					return nil, m.Err()
+				}
+				l, ok := m.(*locatorImpl)
+				if ok {
+					masks = append(masks, map[string]interface{}{
+						"selector": l.selector,
+						"frame":    l.frame.channel,
+					})
+				}
+			}
+			overrides["mask"] = masks
+			options[0].Mask = nil
+		}
 	}
-	data, err := p.channel.Send("screenshot", options)
+	data, err := p.channel.Send("screenshot", options, overrides)
 	if err != nil {
 		return nil, fmt.Errorf("could not send message :%w", err)
 	}

--- a/tests/page_test.go
+++ b/tests/page_test.go
@@ -77,6 +77,36 @@ func TestPageScreenshot(t *testing.T) {
 	require.NoError(t, err)
 }
 
+func TestPageScreenshotWithMask(t *testing.T) {
+	BeforeEach(t)
+	defer AfterEach(t)
+
+	require.NoError(t, page.SetContent("<h1>foobar</h1><p>sensitive</p>"))
+	tmpfile, err := os.MkdirTemp("", "screenshot")
+	require.NoError(t, err)
+	screenshotPath := filepath.Join(tmpfile, "image.png")
+	screenshot1, err := page.Screenshot()
+	require.NoError(t, err)
+	require.True(t, filetype.IsImage(screenshot1))
+	require.Greater(t, len(screenshot1), 50)
+
+	sensElem := page.Locator("p")
+
+	screenshot2, err := page.Screenshot(playwright.PageScreenshotOptions{
+		Path: playwright.String(screenshotPath),
+		Mask: []playwright.Locator{
+			sensElem,
+		},
+	})
+	require.NoError(t, err)
+	require.True(t, filetype.IsImage(screenshot2))
+	require.Greater(t, len(screenshot2), 50)
+	require.NotEqual(t, screenshot1, screenshot2)
+
+	_, err = os.Stat(screenshotPath)
+	require.NoError(t, err)
+}
+
 func TestPagePDF(t *testing.T) {
 	BeforeEach(t)
 	defer AfterEach(t)

